src/import/pull-common.c

```diff
@@ -637,6 +640,89 @@ int pull_job_restart_with_sha256sum(PullJob *j, char **ret) {
         return 1;
 }
 
+int signature_style_from_url(const char *url, SignatureStyle *ret) {
+        _cleanup_free_ char *last = NULL;
+        int r;
+
+        assert(url);
+        assert(ret);
+
+        /* Determines which kind of signature style is appropriate for this url */
+
+        r = import_url_last_component(url, &last);
+        if (r < 0)
+                return r;
+
+        if (streq(last, "SHA256SUMS.gpg")) {
+                *ret = SIGNATURE_GPG_PER_DIRECTORY;
+                return 0;
+        }
+
+        if (streq(last, "SHA256SUMS.asc")) {
+                *ret = SIGNATURE_ASC_PER_DIRECTORY;
+                return 0;
+        }
+
+        if (endswith(last, ".sha256.gpg")) {
+                *ret = SIGNATURE_GPG_PER_FILE;
+                return 0;
+        }
+
+        if (endswith(last, ".sha256.asc")) {
+                *ret = SIGNATURE_ASC_PER_FILE;
+                return 0;
+        }
+
+        return -EINVAL;
+}
+
+int pull_job_restart_with_signature(PullJob *j, char **ret) {
+        SignatureStyle style;
+        int r;
+
+        assert(j);
+
+        /* Generic implementation of a PullJobNotFound handler, that restarts the job requesting a different signature file */
```

Comment:
Maybe also add a note here explaining the order of the fallbacks so that it's clearer